flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1398,22 +1421,43 @@ def test_attach_unknown_volume(self):
             attach_to=self.this_node,
         )
 
-    def test_attach_volume_validate_size(self):
+    def test_size_divisible_by_allocation_unit(self):
         """
-        Validate that attached volume size reported from EBS, OS,
-        and IBlockDeviceAPI layers is consistent.
+        Validate that block devices created by the API, allocate size
+        in intervals of ``allocation_unit`` when the requested size is
+        divisible by the ``allocation_unit``.
         """
-        dataset_id = uuid4()
+        requested_size = allocated_size(
+            self.api.allocation_unit(),
+            REALISTIC_BLOCKDEVICE_SIZE
+        )
 
-        new_volume = self.api.create_volume(
-            dataset_id=dataset_id,
-            size=REALISTIC_BLOCKDEVICE_SIZE
+        self._verify_volume_size(
+            requested_size=requested_size,
+            expected_volume_size=requested_size
         )
-        attached_volume = self.api.attach_volume(
-            new_volume.blockdevice_id, attach_to=self.this_node,
+
+    def test_size_round_up(self):
+        """
+        Validate that block devices created by the API, allocate size
```

Comment:
This puts the burden of rounding on `IBlockDeviceAPI`. I think that's wrong, it means duplicating functionality in all the backends. Not to mention that the deployer (or really, state changes) already does it, there's no need for `IBlockDeviceAPI` implementations to do any rounding.